src/backend/distributed/commands/table.c

```diff
@@ -1439,12 +1432,59 @@ PreprocessAlterTableStmt(Node *node, const char *alterTableCommand,
 				}
 			}
 
+			/*
+			 * We support deparsing for ADD COLUMN only of it's the only
+			 * subcommand.
+			 */
+			if (list_length(commandList) == 1 &&
+				alterTableStatement->objtype == OBJECT_TABLE)
+			{
+				deparseAT = true;
+
+				/*
+				 * Given that we're in the preprocess, any reference to the
+				 * column that we're adding would break the deparser. This
+				 * can only be the case with CHECK constraints. For this
+				 * reason, we don't allow check constraints to be added by
+				 * using ADD COLUMN subcommands. For other constraint types,
+				 * we prepare the constraint to make sure that we can deparse
+				 * it.
+				 *
+				 * TODO: discuss if we're okay with check constraint limitation
```

Comment:
discuss with the team if we're okay with this limitation

Reply:
check if the workaround proposal is applicable for django

Reply:
I feel like we're going down a rabbit hole where we keep putting more stuff (special case logic) into preprocess to work around the fact that a lot of our logic was supposed to be in postprocess.

I think we could tolerate it to fix the Django issue, but it's a bit concerning.